automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/methodconfigs/WorkspaceMethodConfigDetailsPage.scala

```diff
@@ -168,6 +182,19 @@ class WorkspaceMethodConfigDetailsPage(namespace: String, name: String, methodCo
   }
 }
 
+class PopulateFromJsonModal(implicit webDriver: WebDriver) extends OKCancelModal("upload-json-modal") {
+  private val fileInput = FileSelector("data-upload-input" inside this)
+  private val uploadButton = Button("confirm-upload-metadata-button" inside this)
+
+  override def awaitReady(): Unit = fileInput.awaitEnabled()
+
+  def importFile(file: String): Unit = {
+    fileInput.selectFile(file)
+    await condition find(query).get.underlying.getText.contains("Previewing")
+    uploadButton.doClick()
+    await notVisible query
```

Comment:
where does `query` come from?

Reply:
`Component` by way of `OKCancelModal`, so this is checking the modal itself.